### PR TITLE
Updated the new flutter app template to use the default Material themes

### PR DESCRIPTION
**Pull request by darrenaustin**
## Description

This updates the default flutter app that is created with `flutter create` to provide both light and dark themes using the current Material baseline colors schemes for both.

Hopefully this will lead to more apps supporting dark mode out of the box, as well as moving apps toward the latest Material design.

TemplateThemes

The main difference here is that we are using the new `ThemeData.from(colorsScheme)` method that allows us to define a theme from just a `ColorScheme`:

```dart
    return MaterialApp(
      title: 'Flutter Demo',
      // This is the theme of your application.
      //
      // Try running your application with "flutter run". You'll see the
      // application has a purple toolbar. Then, without quitting the app,
      // try changing the primary color of the theme to green by modifying
      // the line below to:
      //
      // theme: ThemeData.from(colorScheme: ColorScheme.light(primary: Colors.green)),
      //
      // and then invoke "hot reload" (press "r" in the console where you ran
      // "flutter run", or simply save your changes to "hot reload" in a Flutter
      // IDE). Notice that the counter didn't reset back to zero; the application
      // is not restarted.
      theme: ThemeData.from(colorScheme: ColorScheme.light()),

      // This is the theme that will be used for you application if the
      // system's 'Dark mode' preference is set.
      darkTheme: ThemeData.from(colorScheme: ColorScheme.dark()),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
```

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/templates/app/lib/main.dart.tmpl [hunk 1]
```diff
@@ -28,18 +28,24 @@ class MyApp extends StatelessWidget {
...
+      // try changing the primary color of the theme to green by modifying
+      // the line below to:
+      //
+      // theme: ThemeData.from(colorScheme: ColorScheme.light(primary: Colors.green)),
+      //
+      // and then invoke "hot reload" (press "r" in the console where you ran
+      // "flutter run", or simply save your changes to "hot reload" in a Flutter
+      // IDE). Notice that the counter didn't reset back to zero; the application
+      // is not restarted.
+      theme: ThemeData.from(colorScheme: ColorScheme.light()),
```
Comment (Hixie) [inline packages/flutter_tools/templates/app/lib/main.dart.tmpl]: I think we need to have this be simpler, e.g.:
```
theme: ThemeData.baseline(
  ColorScheme's arguments
  textTheme
  various other themes like chipTheme et al
),
theme: ThemeData.baselineDark(
  same as above, but different defaults
),
```

That way the change becomes:
```
theme: ThemeData.baseline(primary: Colors.blue),
```
...to:
```
theme: ThemeData.baseline(primary: Colors.green),
```
...and the dark theme is:
```
darkTheme: ThemeData.baselineDark(),
```

File: packages/flutter_tools/templates/app/lib/main.dart.tmpl [hunk 2]
```diff
@@ -28,18 +28,24 @@ class MyApp extends StatelessWidget {
...
-        // changing the primarySwatch below to Colors.green and then invoke
-        // "hot reload" (press "r" in the console where you ran "flutter run",
-        // or simply save your changes to "hot reload" in a Flutter IDE).
-        // Notice that the counter didn't reset back to zero; the application
-        // is not restarted.
-        primarySwatch: Colors.blue,
-      ),
+      // This is the theme of your application.
+      //
+      // Try running your application with "flutter run". You'll see the
```
Comment (RedBrogdon) [inline packages/flutter_tools/templates/app/lib/main.dart.tmpl]: Given that "flutter run" is a command line statement, should it be delimited with backticks?

**Comment by zanderso** [comment]
@darrenaustin @Hixie What is the status of this PR?

**Comment by zanderso** [comment]
Hi @darrenaustin, this PR has been idle for some time, so I'm going to close it, but feel free to open it again if you'd like to continue with it. Thanks!

**Pull request closed without merging**